ext/date/php_date.c

```diff
@@ -4132,15 +4131,30 @@ PHP_METHOD(DatePeriod, __construct)
 	dpobj->current = NULL;
 
 	if (isostr) {
+		zend_replace_error_handling(EH_THROW, zend_ce_value_error, &error_handling);
```

Comment:
I'm wondering if the `ValueError` is ok (BC-wise). Should I change the exception type for the other similar cases as well?

Reply:
In case of `DateTime*::__construct()`, this seems a bit too disruptive

Reply:
Yeah, this was discussed with Derick before, and the outcome was that existing exceptions should stay the same type. I think the change to TypeError above is okay (would normally be thrown by zpp), but this one isn't (catching Exception to validate an ISO date period string would be something that current code can conceivably do).